tests/object.cpp

```diff
@@ -175,4 +277,219 @@ TEST_CASE("object") {
             REQUIRE_INDICES(change.modifications, 0);
         }
     }
+
+    Defaults d;
+    auto create = [&](util::Any&& value, bool update) {
+        r->begin_transaction();
+        auto obj = Object::create(&d, r, *r->schema().find("all types"), value, update);
+        r->commit_transaction();
+        return obj;
+    };
+
+    SECTION("create object") {
+        auto obj = create(AnyDict{
+            {"pk", 1LL},
+            {"bool", true},
+            {"int", 5LL},
+            {"float", 2.2f},
+            {"double", 3.3},
+            {"string", "hello"s},
+            {"data", "olleh"s},
+            {"date", Timestamp(10, 20)},
+            {"object", AnyDict{{"value", 10LL}}},
+            {"array", AnyVec{AnyDict{{"value", 20LL}}}},
+        }, false);
+
+        auto row = obj.row();
+        REQUIRE(row.get_int(0) == 1);
+        REQUIRE(row.get_bool(1) == true);
+        REQUIRE(row.get_int(2) == 5);
+        REQUIRE(row.get_float(3) == 2.2f);
+        REQUIRE(row.get_double(4) == 3.3);
+        REQUIRE(row.get_string(5) == "hello");
+        REQUIRE(row.get_binary(6) == BinaryData("olleh", 5));
+        REQUIRE(row.get_timestamp(7) == Timestamp(10, 20));
+        REQUIRE(row.get_link(8) == 0);
+
+        auto link_target = r->read_group().get_table("class_link target")->get(0);
+        REQUIRE(link_target.get_int(0) == 10);
+
+        auto list = row.get_linklist(9);
+        REQUIRE(list->size() == 1);
+        REQUIRE(list->get(0).get_int(0) == 20);
+    }
+
+    SECTION("create uses defaults for missing values") {
+        d["all types"] = {
+            {"bool", true},
+            {"int", 5LL},
+            {"float", 2.2f},
+            {"double", 3.3},
+            {"string", "hello"s},
+            {"data", "olleh"s},
+            {"date", Timestamp(10, 20)},
+            {"object", AnyDict{{"value", 10LL}}},
+            {"array", AnyVec{AnyDict{{"value", 20LL}}}},
+        };
+
+        auto obj = create(AnyDict{
+            {"pk", 1LL},
+            {"float", 6.6f},
+        }, false);
+
+        auto row = obj.row();
+        REQUIRE(row.get_int(0) == 1);
+        REQUIRE(row.get_bool(1) == true);
+        REQUIRE(row.get_int(2) == 5);
+        REQUIRE(row.get_float(3) == 6.6f);
+        REQUIRE(row.get_double(4) == 3.3);
+        REQUIRE(row.get_string(5) == "hello");
+        REQUIRE(row.get_binary(6) == BinaryData("olleh", 5));
+        REQUIRE(row.get_timestamp(7) == Timestamp(10, 20));
+    }
+
+    SECTION("create throws for missing values if there is no default") {
+        REQUIRE_THROWS(create(AnyDict{
```

Comment:
Is it worth testing that the exception that is thrown is the exception we expect?